libs/libapi/src/libapi/utils.py

```diff
@@ -108,44 +105,39 @@ def try_backfill_dataset_then_raise(
     blocked_datasets: list[str],
     hf_token: Optional[str] = None,
     hf_timeout_seconds: Optional[float] = None,
+    storage_clients: Optional[list[StorageClient]] = None,
 ) -> None:
     """
     Tries to backfill the dataset, and then raises an error.
-
-    Raises the following errors:
-        - [`libcommon.exceptions.DatasetInBlockListError`]
-          If the dataset is in the list of blocked datasets.
     """
-    dataset_orchestrator = DatasetOrchestrator(dataset=dataset, blocked_datasets=blocked_datasets)
-    if not dataset_orchestrator.has_some_cache():
-        # We have to check if the dataset exists and is supported
-        try:
-            revision = get_dataset_git_revision(
-                dataset=dataset,
-                hf_endpoint=hf_endpoint,
-                hf_token=hf_token,
-                hf_timeout_seconds=hf_timeout_seconds,
-            )
-        except Exception as e:
-            # The dataset is not supported
-            raise ResponseNotFoundError("Not found.") from e
-        # The dataset is supported, and the revision is known. We set the revision (it will create the jobs)
-        # and tell the user to retry.
-        logging.info(f"Set orchestrator revision for dataset={dataset}, revision={revision}")
-        dataset_orchestrator.set_revision(
-            revision=revision, priority=Priority.NORMAL, error_codes_to_retry=[], cache_max_days=cache_max_days
-        )
+    if has_pending_ancestor_jobs(dataset=dataset, processing_step_names=processing_step_names):
+        logging.debug("Cache entry not found but some jobs are still in progress, so it could exist in the future")
         raise ResponseNotReadyError(
             "The server is busier than usual and the response is not ready yet. Please retry later."
         )
-    elif dataset_orchestrator.has_pending_ancestor_jobs(processing_step_names=processing_step_names):
-        # some jobs are still in progress, the cache entries could exist in the future
+    logging.debug("No pending job that could create the expected cache entry")
+    if has_some_cache(dataset=dataset):
+        logging.debug(
+            "Some cache entries exist, so the dataset is supported, but that cache entry will never be created"
```

Comment:
Maybe it's a dummy question, but I don't understand why that cache entry will never be created?

Reply:
At this point, we know that:
- some cache entries exist for the dataset,
- but there is no pending jobs for the specific step, or for any of its ancestors,
so: we're not in the situation where the cache entry is being created. Even if other jobs exist, they will not create the cache entry we need.